src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2022,16 +2045,232 @@ private IOperation PeekOperand()
+                                        Debug.Assert(visitedArguments[argumentPlaceholder.ArgumentIndex] is IFlowCaptureReferenceOperation or IDiscardOperation);
+                                        AddPlaceholder(argumentPlaceholder, visitedArguments[argumentPlaceholder.ArgumentIndex]);
+                                    }
+                                    placeholderKeys.Add(argumentPlaceholder);
+                                    break;
+                            }
+                        }
+                    }
+
+                    IOperation visitedArgument = VisitRequired(originalArgument);
+                    Debug.Assert(visitedArgument is IFlowCaptureReferenceOperation);
+                    visitedArguments.Add(visitedArgument);
+
+                    foreach (var key in placeholderKeys)
+                    {
+                        RemovePlaceholder(key);
+                    }
+
+                    placeholderKeys.Clear();
+
+                    var nextIndex = i + 1;
+                    if (nextIndex == arguments.Length || !hasInterpolatedStringHandlerArgument(arguments.AsSpan()[nextIndex..]))
+                    {
+                        i = nextIndex;
+                        return true;
+                    }
+
+                    return false;
+                }
+            }
+
+            static bool hasInterpolatedStringHandlerArgument(ReadOnlySpan<IArgumentOperation> arguments)
+            {
+                return arguments.Any(
+                    static arg => arg.Value switch
+                    {
+                        IInterpolatedStringHandlerCreationOperation { HandlerCreation: IObjectCreationOperation { Arguments: { } creationArgs } } =>
+                            hasPlaceholders(creationArgs, UnwrapArgument),
+                        IInterpolatedStringHandlerCreationOperation { HandlerCreation: IDynamicObjectCreationOperation { Arguments: { } creationArgs } } =>
+                            hasPlaceholders(creationArgs, Identity),
+                        IInterpolatedStringHandlerCreationOperation { HandlerCreation: InvalidOperation { Children: { } creationArgs } } =>
+                            hasPlaceholders(creationArgs, Identity),
+                        _ => false
+                    });
+
+                static bool hasPlaceholders<T>(ImmutableArray<T> args, Func<T, IOperation> unwrapper)
+                {
+                    return args.Any(
+                        static (arg, unwrapper) => unwrapper(arg) switch
+                {
+                    IInterpolatedStringHandlerArgumentPlaceholderOperation { PlaceholderKind: not InterpolatedStringArgumentPlaceholderKind.TrailingValidityArgument } => true,
+                    IConversionOperation
+                    {
+                        Operand: IInterpolatedStringHandlerArgumentPlaceholderOperation { PlaceholderKind: not InterpolatedStringArgumentPlaceholderKind.TrailingValidityArgument }
```

Comment:
>IInterpolatedStringHandlerArgumentPlaceholderOperation [](http://example.com/codeflow?start=33&length=54)

This looks fragile. I think when user-defined conversions are involved, original operand might not be the immediate operand of a top level conversion.